linq4j/src/test/java/org/apache/calcite/linq4j/test/ExpressionTest.java

```diff
@@ -883,6 +887,28 @@ public class ExpressionTest {
             Expressions.constant(Linq4jTest.emps)));
   }
 
+  @Test void testWriteRecordConstant(@TempDir Path tempDir) {
```

Comment:
if this is the test for this issue, you may want a corresponding JavaDoc. there are plenty of examples.